test/lib/jdk/test/lib/hprof/parser/Reader.java

```diff
@@ -151,9 +151,8 @@ public static String getStack(String heapFile, int debugLevel)
                 String deCompressedFile = "heapdump" + System.currentTimeMillis() + ".hprof";
                 File out = new File(deCompressedFile);
                 // Decompress to get dump file.
-                try {
-                    GZIPInputStream gis = new GZIPInputStream(new FileInputStream(heapFile));
-                    FileOutputStream fos = new FileOutputStream(out);
+                try (GZIPInputStream gis = new GZIPInputStream(new FileInputStream(heapFile));
```

Comment:
GZipInutStream close() doesn't say anything about closing the original stream, so I think you need to expand this so that you first create the FIS, then the GZIPIS then the FOS.